airbyte-webapp/src/components/connectorBuilder/Builder/StreamConfigView.tsx

```diff
@@ -151,6 +109,70 @@ export const StreamConfigView: React.FC<StreamConfigViewProps> = ({ streamNum, h
   );
 };
 
+const StreamControls = ({
+  streamNum,
+  selectedTab,
+  setSelectedTab,
+  streamFieldPath,
+}: {
+  streamNum: number;
+  streamFieldPath: (path: string) => string;
+  setSelectedTab: (tab: "configuration" | "schema") => void;
+  selectedTab: "configuration" | "schema";
+}) => {
+  const { formatMessage } = useIntl();
+  const [field, , helpers] = useField<BuilderStream[]>("streams");
```

Comment:
Splitting out the controls to isolate the re-render to this component instead of re-rendering the full stream config view.